cli/templateinit.go

```diff
@@ -23,29 +27,40 @@ func (*RootCmd) templateInit() *clibase.Cmd {
 			if err != nil {
 				return err
 			}
-			exampleNames := []string{}
-			exampleByName := map[string]codersdk.TemplateExample{}
+
+			optsToID := map[string]string{}
 			for _, example := range exampleList {
 				name := fmt.Sprintf(
 					"%s\n%s\n%s\n",
 					cliui.Styles.Bold.Render(example.Name),
 					cliui.Styles.Wrap.Copy().PaddingLeft(6).Render(example.Description),
 					cliui.Styles.Keyword.Copy().PaddingLeft(6).Render(example.URL),
 				)
-				exampleNames = append(exampleNames, name)
-				exampleByName[name] = example
+				optsToID[name] = example.ID
 			}
 
-			_, _ = fmt.Fprintln(inv.Stdout, cliui.Styles.Wrap.Render(
-				"A template defines infrastructure as code to be provisioned "+
-					"for individual developer workspaces. Select an example to be copied to the active directory:\n"))
-			option, err := cliui.Select(inv, cliui.SelectOptions{
-				Options: exampleNames,
-			})
-			if err != nil {
-				return err
+			// If the user didn't specify any template, prompt them to select one.
+			if templateIDArg == "" {
+				opts := keys(optsToID)
+				sort.Strings(opts)
+				_, _ = fmt.Fprintln(inv.Stdout, cliui.Styles.Wrap.Render(
+					"A template defines infrastructure as code to be provisioned "+
+						"for individual developer workspaces. Select an example to be copied to the active directory:\n"))
+				selected, err := cliui.Select(inv, cliui.SelectOptions{
+					Options: sort.StringSlice(keys(optsToID)),
```

Comment:
Already keyed and sorted as `opts`, this seems like a duplicate?

Reply:
Yep, I missed this one. 👍